### PR TITLE
WIP - Add week 5

### PR DESCRIPTION
**Pull request by lbragaglia**
Warning: this PR is a work in progress and it's not ready to be merged: I'll update this branch (and rebase/squash) with the full transcript of the lectures' code as I work on it and complete each lecture.

I opened early this PR to get feedback about the organization of content, for example the split excercises/solutions in lecture 1 and in general the handling of the exercises with a solution provided in the lecture or without a solution (see "flatten" at the end of lecture 1).

### CODE REVIEW

File: src/main/scala/week5/5-1 more functions on lists.sc [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  // Exercises
+  //
+
+  val fruit = List("apples", "oranges", "pears")
+  val nums = List(1, 2, 3)
+  val diag3 = List(List(1, 0, 0), List(0, 1, 0), List(0, 0, 1))
+  val empty = List()
+
+  // Exercise
+  def removeAt[T](n: Int, xs: List[T]) = ???
```
Comment (leanton) [inline src/main/scala/week5/5-1 more functions on lists.sc]: I think we should use solution here...

**Comment by lbragaglia** [comment]
Hi, I've just finished week 5: if you agree, this PR can be merged.

**Comment by leanton** [comment]
looks good to me

**Comment by liufengyun** [comment]
Thanks Lorenzo! Could you please help merge, Anton? I've no laptop with me
currently. Thanks.

**Comment by leanton** [comment]
@liufengyun sure

**Pull request merged**